utils.js

```diff
@@ -4759,11 +4759,87 @@ export async function parseCargoTomlData(cargoTomlFile, simple = false) {
   }
 }
 
-export async function parseCargoData(cargoData) {
+/**
+ * Parse a Cargo.lock file to find components within the Rust project.
+ *
+ * @param {string} cargoLockFile A path to a Cargo.lock file. The Cargo.lock-file path may be used as information for extended attributes, such as manifest based evidence.
+ * @param {boolean} simple Return a simpler representation of the component by skipping extended attributes and license fetch.
+ *
+ * @returns {array} A list of the project's components as described by the Cargo.lock-file.
+ */
+export async function parseCargoData(cargoLockFile, simple = false) {
+  const addPackageToList = (packageList, newPackage, { simple }) => {
+    if (!newPackage) {
+      return;
+    }
+
+    const purl = new PackageURL(
+      "cargo",
+      "",
+      newPackage.name,
+      newPackage.version,
+      null,
+      null
+    ).toString();
+    const component = {
+      type: "library",
+      group: newPackage.group,
+      "bom-ref": purl,
+      purl: purl,
+      name: newPackage.name,
+      version: newPackage.version
+    };
+
+    if (newPackage._integrity) {
+      component.hashes = [
+        {
+          alg: "SHA-384",
+          content: pkg._integrity
+        }
+      ];
+    }
+
+    if (!simple) {
+      // Assign evidence according to CycloneDX's confidence recommendations in section Evidence of:
+      // * https://cyclonedx.org/guides/OWASP_CycloneDX-Authoritative-Guide-to-SBOM-en.pdf
+      // The evidence is deemed to be reliable because Cargo itself generates
+      // the Cargo.lock-file based on the listed dependencies in the
+      // Cargo.toml-file and registry information. So, either we get a direct
+      // dependency (very likely), or a transitive dependency based on
+      // evidence from the package information in the Cargo registry.
+      component.evidence = {
+        identity: {
+          field: "purl",
+          confidence: 0.6,
+          methods: [
+            {
+              technique: "manifest-analysis",
+              confidence: 0.6,
+              value: cargoLockFile
+            }
+          ]
+        }
+      };
```

Comment:
I chose the confidence based on the highest recommendation from [CycloneDX's authoritative guide](https://cyclonedx.org/guides/OWASP_CycloneDX-Authoritative-Guide-to-SBOM-en.pdf), section Evidence. This does contrast the 0.7 for the evidence in the `Cargo.toml`-file.

I don't personally have a strong opinion about the confidence value, but staying within the range of the recommendations seems reasonable. And being on the higher end of the scale seems reasonable because it's information from the manifest's own tool.

Reply:
Can we lower the confidence for cargo.toml to 0.5?

Reply:
Sure! I'll fix as soon as I can.

Reply:
> Can we lower the confidence for cargo.toml to 0.5 since toml-based detection is less precise compared to the lock file?

@prabhu 

I didn't think of it when I first read the comment, but having the `Cargo.toml` evidence being higher does make sense, because those will be dependencies explicitly added by the developer. The components from `Cargo.lock` are the inferred transitive dependencies. Because Cargo both maintain the tool and the registry, transitive dependencies are probably reliable, but probably not as reliable as first hand information from the developer.

This PR currently assigns evidence confidence like this:
| File | Confidence |
|:---|---:|
|`Cargo.toml`|0.7|
|`Cargo.lock`|0.6|

This is how I interpret your request:
| File | Confidence |
|:---|---:|
|`Cargo.toml`|0.5|
|`Cargo.lock`|0.6|

And my current suggestion is:

| File | Confidence (Within recommended range) | Confidence (Assign outside recommended range) |
|:---|---:| ---: |
|`Cargo.toml`|0.6 | 0.7|
|`Cargo.lock`|0.5 | 0.6|

(The alternatives is for whether or not we should place the confidence within the recommended ranges).

As I said, I don't have any strong opinion on this and will defer to your request -- you have way more knowledge about this than I do. I want to make sure I understood the request correctly.

Reply:
@Lucasljungberg This is an interesting way to look at the confidence, unfortunately this is different to how it is implemented for other ecosystems within the tools. The rationale to increase the confidence for a lock file is that we have both toml and the lock file so the evidence has slightly increased for the methods. Let's retain the same logic for cargo as well so 0.5 for toml and 0.6 for lock. We can revisit the confidence calculations in a separate ticket once we start seeing some adoption. Hope this is alright for you.

Reply:
Alright! That sounds good!

Reply:
We can release 10.3.0 once this PR is complete. Thank you!